use hardfork infrastructure in latency

paweljakubas opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->
#2021 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] I have added HF infra


# Comments

<!-- Additional comments or screenshots to attach if any -->
How to run locally now : `stack bench cardano-wallet:latency`


At this moment I have : 
```
latency: /home/pawel/Work/IOHK/cardano-wallet/lib/shelley/test/data/cardano-node-shelley/faucet-addrs/faucet101.addr: openBinaryFile: does not exist (No such file or directory)
```
which is surprising, faucet101.addr should not be tried :thinking: 
<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

paweljakubas committed:
use hardfork infrastructure in latency

KtorZ commented on lib/shelley/bench/Latency.hs:
```diff
@@ -362,3 +401,19 @@ benchWithShelleyServer tracers action = do
+        , PoolConfig {retirementEpoch = Just 100_000}
+        -- This pool should retire, but not within the duration of a test run:
+        , PoolConfig {retirementEpoch = Just 1_000_000}
+        ]
```
I don't think we need to have stake pools at all in the API latency benchmarks, do we?

paweljakubas replied:
indeed, fixed in https://github.com/input-output-hk/cardano-wallet/pull/2046/commits/e527debc14af47fcb558e432b16b443388364d68

KtorZ commented on lib/shelley/bench/Latency.hs:
```diff
@@ -345,11 +364,31 @@ benchWithShelleyServer tracers action = do
+        sendFaucetFundsTo stdoutTextTracer dir addresses
+        let rewards = (,Coin $ fromIntegral oneMillionAda) <$>
+                concatMap genRewardAccounts mirMnemonics
+        moveInstantaneousRewardsTo stdoutTextTracer dir rewards
```
The MIR certificates aren't useful in the latency benchmarks, only the standard faucets are needed.

paweljakubas replied:
indeed, fixed in https://github.com/input-output-hk/cardano-wallet/pull/2046/commits/e527debc14af47fcb558e432b16b443388364d68

paweljakubas committed:
slim cluster warm up

KtorZ committed:
fix HLint suggestion about redundant imports

KtorZ committed:
start local cluster only once in latency benchmarks

  Previously, was started before each bench although, some state was shared in-memory causing issues for the second bench to run.

KtorZ committed:
tentatively attempt to provide 'cardano-cli' in scope for the latency benchmarks
